### PR TITLE
WIP - add getting eth balance changes from internal transfers via traces

### PR DESCRIPTION
**Pull request by JasoonS**
This code is still broken: `pnpm all-transfers 0x89e51fa8ca5d66cd220baed62ed01e8951aa7c40`

It should show a possitive (and fairly large) balance for ETH. But it shows a negative balance instead.

Result currently:
```
Total # of transactions made by account - in: 194 out: 4603493
Total Ether volume in: 9732275655361820237243
  Total Ether volume in (EOA): 9850397609708649684425
  Total Ether volume out (EOA): 78027152044453610763
  Total Ether volume in (internal): 9850397609708649684425
  Total Ether volume out (internal): 76463092673573336601
  Total gas paid: 9814029319337452184243
```

### CODE REVIEW

File: src/all-transfers/index.ts [hunk 1]
```diff
@@ -142,19 +146,49 @@ async function main() {
...
         continue;
       }
 
+      //// Sanity check
+      //assert(tx.from === targetAddress || tx.to == targetAddress, `Invalid trace data, neither from nor to is the target address, from: ${tx.from}, to: ${tx.to}, target: ${targetAddress}`)
+      // if (tx.from !== targetAddress && tx.to !== targetAddress) {
+      //   continue; // We will be able to improve this when we have better join modes.
+      // }
+
       if (tx.from === targetAddress) {
```
Comment (ozgrakkurt) [inline src/all-transfers/index.ts:175]: The top level transaction might be duplicated in the traces as well, not 100% sure about this but I remember it being like that

File: src/all-transfers/index.ts [hunk 2]
```diff
@@ -109,8 +113,8 @@ async function main() {
       // skip invalid logs
       if (
         log == undefined ||
-        log.indexed.length < 2 ||
-        log.body.length < 1 ||
+        log.indexed.length == 2 ||
```
Comment (ozgrakkurt) [inline src/all-transfers/index.ts]: is this intentional? Seems like it changes the logic

**Review by ozgrakkurt — APPROVED** [review]

File: src/all-transfers/index.ts (hunk 2 above)
Comment (JasoonS) [reply src/all-transfers/index.ts]: Yeah, you are right it breaks erc20... But I am focussing on ETH at the moment. I should comment it out temporarily :)

Comment (JasoonS) [reply src/all-transfers/index.ts]: I should do `!= 2` and `!=1`

**Pull request merged**